beacon-chain/slasher/config.go

```diff
@@ -0,0 +1,155 @@
+package slasher
+
+// Config parameters for slashing detection. Validator min and max spans
+// split into chunks of length C for easy detection and lazy loading.
+//
+// A regular pair of min and max spans for a validator look as follows
+// with length = H where H is the amount of epochs worth of history
+// we want to persist for slashing detection.
+//
+//  validator_1_min_span = [2, 2, 2, ..., 2]
+//  validator_1_max_span = [0, 0, 0, ..., 0]
+//
+// After chunking, we obtain chunks of length C. For C = 3:
+//
+//  validator_1_min_span_chunk_0 = [2, 2, 2]
+//  validator_1_max_span_chunk_0 = [2, 2, 2]
+//
+// On disk, we take chunks for K validators, and store them as flat byte slices.
+// For example, if H = 3, C = 3, and K = 3, then we can store 3 validators' chunks as a flat
+// slice as follows:
+//
+//     val0     val1     val2
+//      |        |        |
+//   {     }  {     }  {     }
+//  [2, 2, 2, 2, 2, 2, 2, 2, 2]
+//
+// This is known as 2D chunking, pioneered by the Sigma Prime team here:
+// https://hackmd.io/@sproul/min-max-slasher
+//
+// To properly access the element at epoch `e` for a validator index `i`, we leverage helper
+// functions from config values as a nice abstraction. the following parameters are
+// required for the helper functions defined in this file.
+//
+// ChunkSize defines how many elements are in a chunk for a validator
+// min or max span slice.
+// ValidatorChunkSize defines how many validators' chunks we store in a single
+// flat byte slice on disk.
+// HistoryLength defines how many epochs we keep of min or max spans.
+type Config struct {
+	ChunkSize          uint64
+	ValidatorChunkSize uint64
+	HistoryLength      uint64
+}
+
+// DefaultConfig defines default values for slasher's configuration, defined
+// based on optimization analysis for best and worst case scenarios for
+// slasher's performance.
+func DefaultConfig() *Config {
+	return &Config{
+		// The default values for ChunkSize and ValidatorChunkSize were
+		// decided after an optimization analysis performed by the Sigma Prime team.
+		// See: https://hackmd.io/@sproul/min-max-slasher#1D-vs-2D for more information.
+		// We decide to keep 4096 epochs worth of data in each validator's min max spans.
```

Comment:
Unsure about why 4096 and not WEAK_SUBJECTIVITY_PERIOD. Will ask sigp about this